client/etcd/config.go

```diff
@@ -26,16 +26,28 @@ import (
 	"github.com/m3db/m3x/instrument"
 )
 
-// ClusterConfig is the config for a zoned etcd cluster
+// ClusterConfig is the config for a zoned etcd cluster.
 type ClusterConfig struct {
-	Zone      string   `yaml:"zone"`
-	Endpoints []string `yaml:"endpoints"`
-	Cert      string   `yaml:"cert",omitempty`
-	Key       string   `yaml:"key",omitempty`
-	CA        string   `yaml:"ca",omitempty`
+	Zone      string     `yaml:"zone"`
+	Endpoints []string   `yaml:"endpoints"`
+	Auth      AuthConfig `yaml:"auth"`
 }
 
-// Configuration is for config service client
+// AuthConfig is the config for authentication.
+type AuthConfig struct {
+	Cert string `yaml:"cert"`
```

Comment:
A few naming nits:

- rename to TLSConfig
- change var names to reflect what they are (path) and the type of file (crt/key).

Something like this:

```
type TLSConfiguration struct {
  KeyPath string `yaml...`
  CrtPath ... 
  CACrtPath .. 
}```

Reply:
sg